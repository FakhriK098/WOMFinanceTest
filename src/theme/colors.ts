export const colors = {
  primary: '#4F46E5',
  danger: '#DC2626',
  white: '#FFFFFF',
  black: '#000000',
  text: {
    light: '#111827',
    dark: '#F9FAFB',
    mutedLight: '#6B7280',
    mutedDark: '#9CA3AF',
  },
  background: {
    light: '#FFFFFF',
    dark: '#0B1220',
  },
  border: {
    light: '#E5E7EB',
    dark: '#374151',
  },
  inputBg: {
    light: '#1118270D',
    dark: '#111827',
  },
  overlay: 'rgba(0,0,0,0.2)',
}
