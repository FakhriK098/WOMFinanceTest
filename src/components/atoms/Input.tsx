import React, { useState } from 'react'
import { TextInput, StyleSheet, View, useColorScheme } from 'react-native'
import type { InputAtomProps } from '../../types/components'
import { colors } from '../../theme/colors'
import { spacing } from '../../theme/spacing'
import { radius } from '../../theme/radius'
import { dimensions } from '../../theme/dimensions'

const Input = ({ style, error, secureTextEntry, ...rest }: InputAtomProps) => {
  const isDark = useColorScheme() === 'dark'
  const [focused, setFocused] = useState(false)
  return (
    <View style={[
      styles.wrapper,
      { backgroundColor: isDark ? colors.inputBg.dark : colors.inputBg.light, borderColor: isDark ? colors.border.dark : colors.border.light },
      focused && styles.focused,
      error && styles.error,
    ]}>
      <TextInput
        {...rest}
        style={[styles.input, { color: isDark ? colors.text.dark : colors.text.light }, style]}
        placeholderTextColor={isDark ? colors.text.mutedDark : colors.text.mutedLight}
        onFocus={e => {
          setFocused(true)
          rest.onFocus && rest.onFocus(e)
        }}
        onBlur={e => {
          setFocused(false)
          rest.onBlur && rest.onBlur(e)
        }}
        secureTextEntry={secureTextEntry}
        autoCapitalize="none"
      />
    </View>
  )
}

const styles = StyleSheet.create({
  wrapper: {
    borderWidth: 1,
    borderRadius: radius.md,
    paddingHorizontal: spacing.md,
  },
  focused: {
    borderColor: colors.primary,
  },
  error: {
    borderColor: colors.danger,
  },
  input: {
    height: dimensions.controlHeight,
  },
})

export default React.memo(Input)
