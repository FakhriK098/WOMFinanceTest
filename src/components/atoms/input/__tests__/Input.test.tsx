import React from 'react';
import renderer, { act } from 'react-test-renderer';

import Input from '../Input';
import { colors } from '@theme/colors';

describe('Input atom', () => {
  beforeEach(() => {
    jest.resetModules();
  });

  it('matches snapshot', () => {
    const RN = require('react-native');
    jest.spyOn(RN, 'useColorScheme').mockReturnValue('light');
    let tr: renderer.ReactTestRenderer;
    act(() => {
      tr = renderer.create(
        <Input value="" onChangeText={() => {}} placeholder="Type" />,
      );
    });
    expect(tr!.toJSON()).toMatchSnapshot();
    act(() => tr!.unmount());
  });

  it('propagates secureTextEntry', () => {
    const RN = require('react-native');
    jest.spyOn(RN, 'useColorScheme').mockReturnValue('light');
    let tr: renderer.ReactTestRenderer;
    act(() => {
      tr = renderer.create(<Input value="" onChangeText={() => {}} secureTextEntry />);
    });
    const ti = tr!.root.find((inst) => inst.props && inst.props.placeholderTextColor);
    expect(ti.props.secureTextEntry).toBe(true);
    act(() => tr!.unmount());
  });

  it('adds focused border color on focus', () => {
    const RN = require('react-native');
    jest.spyOn(RN, 'useColorScheme').mockReturnValue('light');
    let tr: renderer.ReactTestRenderer;
    act(() => {
      tr = renderer.create(<Input value="" onChangeText={() => {}} />);
    });
    const input = tr!.root.find((inst) => inst.props && typeof inst.props.onFocus === 'function');
    act(() => {
      input.props.onFocus({} as any);
    });
    const wrapper = tr!.root.find((inst) => inst.props && Array.isArray(inst.props.style));
    const style = wrapper.props.style as any[];
    const hasPrimary = style.some((s) => s && s.borderColor === colors.primary);
    expect(hasPrimary).toBe(true);
    act(() => tr!.unmount());
  });

  it('applies error border color when error=true', () => {
    const RN = require('react-native');
    jest.spyOn(RN, 'useColorScheme').mockReturnValue('light');
    let tr: renderer.ReactTestRenderer;
    act(() => {
      tr = renderer.create(<Input value="" onChangeText={() => {}} error />);
    });
    const wrapper = tr!.root.find((inst) => inst.props && Array.isArray(inst.props.style));
    const style = wrapper.props.style as any[];
    const hasDanger = style.some((s) => s && s.borderColor === colors.danger);
    expect(hasDanger).toBe(true);
    act(() => tr!.unmount());
  });
});
