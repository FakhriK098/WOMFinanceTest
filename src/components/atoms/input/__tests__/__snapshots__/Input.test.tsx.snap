// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input atom matches snapshot 1`] = `
<View
  style={
    [
      {
        "borderRadius": 12,
        "borderWidth": 1,
        "paddingHorizontal": 14,
      },
      {
        "backgroundColor": "#1118270D",
        "borderColor": "#E5E7EB",
      },
      false,
      undefined,
    ]
  }
>
  <TextInput
    autoCapitalize="none"
    onBlur={[Function]}
    onChangeText={[Function]}
    onFocus={[Function]}
    placeholder="Type"
    placeholderTextColor="#6B7280"
    style={
      [
        {
          "height": 48,
        },
        {
          "color": "#111827",
        },
        undefined,
      ]
    }
    value=""
  />
</View>
`;
